src/pe-employee-recognition-nominate/widget/pe-employee-recognition-nominate.u-server-script.js

```diff
@@ -1,4 +1,89 @@
 (function() {
-  /* populate the 'data' object */
-  /* e.g., data.table = $sp.getValue('table'); */
+	
+	data.msgs = {
+		success: gs.getMessage("Thank you for nominating."),
+		error: gs.getMessage("Something didn't work."),
+		no_value: gs.getMessage("No selection was detected."),
+		your_selection: gs.getMessage("Your selection"),
+		my_candidates: gs.getMessage("My Candidates")
+	}
+	
+	var ucl = new sn_uc.UserCriteriaLoader();
+	var currentUserCriteria = ucl.getAllUserCriteria();
+	
+	var scope_prefix = 'x_snc_empra_';
+	var vote_cycle_table = scope_prefix + 'cycle';
+	var vote_option_table = scope_prefix + 'candidate';
+	var vote_cast_table = scope_prefix + 'vote_cast';
+	
+	if (input && input.action == 'nominate') {
+		data.responses = {};
+		var newNomGR = new GlideRecord(vote_option_table);
+		newNomGR.initialize();
+		newNomGR.user = input.candidate.value;
+		newNomGR.name = input.candidate.displayValue;
+		newNomGR.voting_cycle = input.voting_cycle.id;
+		newNomGR.achievement = input.achievement;
+		newNomGR.short_description = 'Nomination: '+input.candidate.displayValue+' for '+input.voting_cycle.name;
+		newNomGR.approval = 'requested';
+		if (newNomGR.insert())
+			data.success = true;
+		return;
+	}
+  
+	var cycles = [];
+	
+	var gr = new GlideRecord(vote_cycle_table);
+	gr.addQuery('state','1');
+	gr.query();
+	
+	while (gr.next()) {
+		var nominatorCriteriaId = gr.nominator_criteria.toString();
+		if ( nominatorCriteriaId && currentUserCriteria.indexOf(nominatorCriteriaId) > -1 ) continue;
+			
+		var myCandidates = [];
+		var myCandidateIds = [];
+
+		var cGR = new GlideRecord(vote_option_table);
+		cGR.addQuery('opened_by', gs.getUserID());
```

Comment:
consider making the variable name more meaningful. 

https://medium.com/coding-skills/clean-code-101-meaningful-names-and-functions-bf450456d90c